sdk/python/test/e2e/test_e2e_pytorchjob.py

```diff
@@ -53,103 +59,136 @@ def test_sdk_e2e_with_gang_scheduling(job_namespace):
 
     master = KubeflowOrgV1ReplicaSpec(
         replicas=1,
         restart_policy="OnFailure",
-        template=V1PodTemplateSpec(metadata=V1ObjectMeta(annotations={constants.ISTIO_SIDECAR_INJECTION: "false"}),
-                                   spec=V1PodSpec(containers=[container])),
+        template=V1PodTemplateSpec(
+            metadata=V1ObjectMeta(
+                annotations={constants.ISTIO_SIDECAR_INJECTION: "false"}
+            ),
+            spec=V1PodSpec(containers=[container]),
+        ),
     )
 
     worker = KubeflowOrgV1ReplicaSpec(
         replicas=1,
         restart_policy="OnFailure",
-        template=V1PodTemplateSpec(metadata=V1ObjectMeta(annotations={constants.ISTIO_SIDECAR_INJECTION: "false"}),
-                                   spec=V1PodSpec(containers=[container])),
+        template=V1PodTemplateSpec(
+            metadata=V1ObjectMeta(
+                annotations={constants.ISTIO_SIDECAR_INJECTION: "false"}
+            ),
+            spec=V1PodSpec(containers=[container]),
+        ),
     )
 
-    pytorchjob = generate_pytorchjob(master, worker, job_namespace=job_namespace)
+    pytorchjob = generate_pytorchjob(job_namespace, master, worker)
 
-    TRAINING_CLIENT.create_pytorchjob(pytorchjob, job_namespace)
-    logging.info(f"List of created {constants.PYTORCHJOB_KIND}s")
-    logging.info(TRAINING_CLIENT.list_pytorchjobs(job_namespace))
+    TRAINING_CLIENT.create_job(job=pytorchjob, namespace=job_namespace)
+    logging.info(f"List of created {TRAINING_CLIENT.job_kind}s")
+    logging.info(TRAINING_CLIENT.list_jobs(job_namespace))
 
-    verify_job_e2e(
-        TRAINING_CLIENT,
-        JOB_NAME,
-        job_namespace,
-        constants.PYTORCHJOB_KIND,
-        CONTAINER_NAME,
-        timeout=900,
+    verify_job_e2e(TRAINING_CLIENT, JOB_NAME, job_namespace, timeout=900)
+
+    TRAINING_CLIENT.delete_job(JOB_NAME, job_namespace)
+
+
+def test_pytorchjob_from_func(job_namespace):
+    # Test Training function.
+    def train_func(parameters):
+        import pandas as pd
+        import time
+
+        print(f"Package pandas=={pd.__version__} is installed")
+        print(f"Input function parameters are: {parameters}")
+
+        print("Stat Training ....")
+        for i in range(10):
+            print(f"Epoch: {i} finished")
+            time.sleep(1)
+
+        print("Training is complete")
+
+    TRAINING_CLIENT.create_job(
+        name=JOB_NAME,
+        namespace=job_namespace,
+        parameters={"lr": "0.01"},
+        train_func=train_func,
+        num_worker_replicas=1,
+        packages_to_install=["pandas==1.3.5"],
     )
 
-    TRAINING_CLIENT.delete_pytorchjob(JOB_NAME, job_namespace)
+    logging.info("Get created PyTorchJob from function")
+    logging.info(TRAINING_CLIENT.get_job(JOB_NAME, job_namespace))
+
+    verify_job_e2e(TRAINING_CLIENT, JOB_NAME, job_namespace, timeout=900)
+    TRAINING_CLIENT.delete_job(JOB_NAME, job_namespace)
```

Comment:
Uhm, I think this test is for SDK, not the operator. So ideally, we should put this test on unit tests.
However, I'm not sure if we can put this test on unit tests.
@andreyvelich WDYT?

Reply:
I was thinking that it might be better to run this test case as unit tests if there is a fake client for Python like [client-go](https://github.com/kubernetes/client-go/blob/908cc0a33323050d5a51c4423a1a038fffbce318/kubernetes/fake/clientset_generated.go).

However, I could not find such a fake client for Python. So we should keep having this case in e2e.

Reply:
SGTM